### PR TITLE
Make change_message_visibility public

### PR DESCRIPTION
**Pull request by bennapp**
We would like to exit processes and re-queue the messages they had before they exit.

We are doing this in order to support our deployment process.

### CODE REVIEW

File: lib/circuitry/subscriber.rb [hunk 1]
```diff
@@ -84,6 +84,11 @@ def lock=(value)
       @lock = value
     end
 
+    def change_message_visibility(message)
+      logger.info("Retrying message now by making the 'visiblity_timeout' zero seconds for message #{message.id}")
+      sqs.change_message_visibility(queue_url: queue, receipt_handle: message.receipt_handle, visibility_timeout: 0)
+    end
+
```
Comment (diclophis) [inline lib/circuitry/subscriber.rb:91]: Are we sure there isn't some built in way to return / raise an exception that does this automagically... ie.

```
def run
  circuit.subscribe do |msg|
    # possibly something like this
    raise Circuit::Renequeue.new

    # or possibly something like
    return :reenqueue

    # or some other bit to signal circuitry to re-enqueue immediatly?
end
```

I feel like maybe this was private for a reason??? maybe not, it is ruby after all

**Review by diclophis — APPROVED** [review]
see comment, please reply, LGTM

File: lib/circuitry/subscriber.rb (hunk 1 above)
Comment (bennapp) [reply lib/circuitry/subscriber.rb:91]: @diclophis We added this method to circuitry; so I am comfortable making it public. Come to think of it, we are using this method as such:

```
rescue => e
      change_message_visibility(message) if ignore_visibility_timeout
...
```

and we are setting: `ignore_visibility_timeout` to `true`.

What this means is we could have changed this flow to:
1. Raise in our primary application (bigmaven)
2. Circuitry would catch that exception and re-queue the message.

I am fine with what we are doing:
1. Re-queueing the message
2. Exiting without raising an exception (and thus no errors in our logs etc.)

Comment (diclophis) [reply lib/circuitry/subscriber.rb:91]: `msg.retry!` type thng?

**Review by bmidgley — APPROVED** [review]

**Comment by diclophis** [comment]
right right

**Pull request merged**